retranmission.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// MessageStore represents the retransmission subsystem storage for lnwire
+// messages, it might be boltd storage, local storage, test storage or hybrid
+// one.
+//
+// NOTE: The original purpose of creating this interface was in separation
+// between the retransmission logic and specifics of storage itself. In case of
+// such interface we may create the test storage and register the add,remove,get
+// actions without the need of population of lnwire messages with data.
+type MessageStore interface {
+	// Get returns the sorted set of messages in the order they were
+	// added in storage.
+	Get() ([]lnwire.Message, error)
+
+	// Add adds new message in the storage with preserving the order.
+	Add(msg lnwire.Message) error
+
+	// Remove deletes all messages that corresponds to codes from storage.
+	//
+	// Note: Why this function gets set of codes as input?
+	// Is due to the fact of logic of retransmission subsystem, where we
+	// need remove messages not one by one but in contrary by groups of
+	// messages.
+	Remove(codes ...lnwire.Code) error
```

Comment:
With the modification to the schema I suggested, I think this method would be changed to something along the lines of an `Ack` method and instead take a single `lnwire.MessageCode`.